packages/design-tokens-schema/src/remove-non-token-properties.ts

```diff
@@ -0,0 +1,50 @@
+import { BaseDesignTokenProperties } from './base-token';
+import { isTokenLike } from './token-reference';
+
+const knownKeys = new Set<string>(BaseDesignTokenProperties);
+
+const isRecord = (value: unknown): value is Record<string, unknown> => {
+  return typeof value === 'object' && value !== null && !Array.isArray(value);
+};
+
+/**
+ * Internal function that handles non-record inputs
+ */
+const processValue = (value: unknown): unknown => {
+  if (typeof value !== 'object' || value === null) {
+    return value;
+  }
+
+  if (Array.isArray(value)) {
+    return value.map(processValue);
+  }
+
+  const cleaned: Record<string, unknown> = Object.create(null);
+
+  for (const [key, val] of Object.entries(value)) {
+    // For tokens, skip any non-standard keys (metadata)
+    // For non-tokens, keep all keys but recursively strip their values
+    if (isTokenLike(value) && !knownKeys.has(key)) {
```

Comment:
I was thrown for a sec since I did not expect a Set. I was expecting something like this:

```suggestion
    if (isTokenLike(value) && !baseDesignTokenProperties.includes(key)) {
```

What is the benefit of doing this with a Set instead of an Array?